addons/web_editor/controllers/main.py

```diff
@@ -682,11 +698,97 @@ def shape(self, module, filename, **kwargs):
+                                   svg: str):
+        """
+        Replace animation durations in SVG and CSS with modified values.
+
+        This function takes a speed value and an SVG string containing
+        animations. It uses regular expressions to find and replace the
+        duration values in both CSS animation rules and SVG duration attributes
+        based on the provided speed.
+
+        Parameters:
+            - speed (float): The speed used to calculate the new animation
+            durations.
+            - svg (str): The SVG string containing animations.
+
+        Returns:
+        str: The modified SVG string with updated animation durations.
+        """
+        ratio = (1 + shape_animation_speed
+                 if shape_animation_speed >= 0
+                 else 1 / (1 - shape_animation_speed))
+
+        def callback_css_animation_rule(match):
+            # Extracting matched groups.
+            declaration, value, unit, separator = (
+                match.group("declaration"),
+                match.group("value"),
+                match.group("unit"),
+                match.group("separator"),
+            )
+            # Calculating new animation duration based on ratio.
+            value = str(float(value) / (ratio or 1))
+            # Constructing and returning the modified CSS animation rule.
+            return f"{declaration}{value}{unit}{separator}"
+
+        def callback_svg_dur_timecount_val(match):
+            attribute_name, value, unit = (
+                match.group("attribute_name"),
+                match.group("value"),
+                match.group("unit"),
+            )
+            # Calculating new duration based on ratio.
+            value = str(float(value) / (ratio or 1))
+            # Constructing and returning the modified SVG duration attribute.
+            return f'{attribute_name}{value}{unit or "s"}"'
+
+        def callback_css_animation_ratio(match):
+            ratio = match.group("ratio")
+            return f'--animation_ratio: {ratio};'
+
+        # Applying regex substitutions to modify animation speed in the
+        # 'svg' variable.
+        svg = re.sub(
+            CSS_ANIMATION_RULE_REGEX,
+            callback_css_animation_rule,
+            svg
+            )
+        svg = re.sub(
+            SVG_DUR_TIMECOUNT_VAL_REGEX,
+            callback_svg_dur_timecount_val,
+            svg
+            )
+        # Create or modify the css variable --animation_ratio for future
+        # purpose.
+        if re.match(CSS_ANIMATION_RATIO_REGEX, svg):
+            svg = re.sub(
+                CSS_ANIMATION_RATIO_REGEX,
+                callback_css_animation_ratio,
+                svg
+            )
+        else:
+            regex = r"<svg .*>"
+            declaration = f"--animation ratio: {ratio}"
+            subst = ("\\g<0>\n\t<style>\n\t\t:root { \n\t\t\t" +
+                     declaration +
+                     ";\n\t\t}\n\t</style>")
```

Comment:
Can't you rather add a `style="--animation-ratio: ..."` on the SVG node itself? Also you wrote `animation ratio` instead of `animation_ratio` which you used elsewhere... normal? :thinking: